analyzer/src/main/kotlin/managers/utils/NuGetServiceIndex.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2022 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.analyzer.managers.utils
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+
+// See https://docs.microsoft.com/en-us/nuget/api/service-index.
+@JsonIgnoreProperties(ignoreUnknown = true)
+data class ServiceIndex(
```

Comment:
I've investigated the data classes in the original file.
To me it seems that all the classes in `NuGetAllPackageData.kt` except for the 
top-level class do represent the NuGet data model and are used for de-serialization,
while the top-level one is something different: it's a data model of Ort's Nuget implementation.

I believe that separating model classes while keeping the non-model class seems to not extract the things which belongs there the least, while it extracts two classes which actually belong to the model. Also the thing which gets extracted is similar close related as the classes which remain in the file. Does this make any sense?

How about first doing the following?

1. move the top level class `NuGetAllPackageData` into it's only user. IMO it is similar to `NuGetDependency` and belongs next to it.
2. rename the file to NuGetModel
and Note: The result would be no more nesting of classes in this file.

Given that state, maybe it suffices to keep the classes in a single file
and structure / nest them as follows? 
```
PackageData

PackageDetails
  DependencyGroup
  Dependency

PackageSpec
  Metadata
  License
  Repository

ServiceIndex
  ServiceResouce -> rename to Resource?
```

Reply:
> all classses [...] represent the NuGet data model and are used for de-serialization, while the top-level one is something different: it's a data model of Ort's Nuget implementation.

That's correct.

> Also the thing which gets extracted is similar close related as the classes which remain in the file. Does this make any sense?

I view it a bit differently (and I've reworded the commit message accordingly): My goal in that commit was to separate classes that are used for deserialization, but that deserialize completely different things. One set of classes deserializes information about where to get package metadata *from*, whereas the other set of classes deserializes the actual package metadata.

I can see the point in further separating classes used directly for deserialization from classes that just bundle / make use of these classes and will take a look at that.

PS: About "separating model classes while keeping the non-model class", to me "model classes" are not only those used directly for deserialization, but any (data) classes that represent the data model, including classes that just bundle the classes used for deserialization.

Reply:
> PS: About "separating model classes while keeping the non-model class", to me "model classes" are not only those used directly for deserialization, but any (data) classes that represent the data model, including classes that just bundle the classes used for deserialization.

Well, when I used the term "model classes" it seems it wasn't sufficiently clear.

I wanted to propose to clearly separate the following:
1. Classes which represent upstream data model (these are the ones which I referred to as "model" classes)
2. Classes which represent ORT's custom data model

Reply:
Yes, that new commit is what I actually thought would be needed in order to satisfy this one, thanks!